neo4j-cypher-dsl-schema-name-support/src/test/java/org/neo4j/cypherdsl/support/schema_name/SchemaNamesIT.java

```diff
@@ -0,0 +1,142 @@
+ * limitations under the License.
+ */
+package org.neo4j.cypherdsl.support.schema_name;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.DynamicContainer;
+import org.junit.jupiter.api.DynamicNode;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Result;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.summary.ResultSummary;
+import org.neo4j.driver.types.Node;
+import org.testcontainers.containers.Neo4jContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+/**
+ * @author Michael J. Simons
+ */
+@Testcontainers(disabledWithoutDocker = true)
+class SchemaNamesIT {
+
+	private static final Config DRIVER_CONFIG = Config.builder().withMaxConnectionPoolSize(1).build();
+	private static Map<Neo4jContainer<?>, Driver> CACHED_CONNECTIONS = Collections.synchronizedMap(new HashMap<>());
+	private static String LATEST_VERSION = "4.4";
+
+	String[][] TEST_DATA = new String[][] {
+		{ "ABC", "ABC" },
+		{ "A C", "A C" },
+		{ "A` C", "A` C" },
+		{ "A`` C", "A` C" },
+		{ "ALabel", "ALabel" },
+		{ "A Label", "A Label" },
+		{ "A `Label", "A `Label" },
+		{ "`A `Label", "`A `Label" },
+		{ "Spring Data Neo4j⚡️RX", "Spring Data Neo4j⚡️RX" },
+		{ "`", "`" },
+		{ "\u0060", "`" },
+		{ "```", "``" },
+		{ "\u0060\u0060\u0060", "``" },
+		{ "\\u0060\\u0060\\u0060", "``" },
+		{ "Hello`", "Hello`" },
+		{ "Hi````there", "Hi``there" },
+		{ "Hi`````there", "Hi```there" },
+		{ "`a`b`c`", "`a`b`c`" },
+		{ "\u0060a`b`c\u0060d\u0060", "`a`b`c`d`" },
+		{ "\\u0060a`b`c\\u0060d\\u0060", "`a`b`c`d`" },
+		{ "Foo\\\\`bar", "Foo\\`bar" },
+		// Escape the unicode literal for java
+		{ "admin\\user", "admin\\user" },
+		// Escape the unicode literal for cypher
+		{ "admin\\\\user", "admin\\user" },
+		// The unicode escaped
+		{ "\\u0075\\u1456", "uᑖ" },
+		// Not supported, "Potential additional rules mentioned in the GQL standard that would be potentially worth considering could take over:"
+		// new TestInput("\\u000151", "ő", "ő"),
+		{ "\u1456", "ᑖ" },
+		{ "something\\u005C\\u00751456", "something\\u1456" },
+		{ "x\\y", "x\\y" },
+		// Already escaped backslash
+		{ "x\\\\y", "x\\y" },
+		// Escaped backticks
+		{ "x\\`y", "x`y" },
+		{ "x\\```y", "x``y" },
+		{ "x`\\```y", "x```y" },
+		// This is the backtick itself in the string
+		{ "Foo \u0060", "Foo `" },
+		// This is the backtick unicode escaped so that without further processing `foo \u0060` would end up at Cypher,
+		{ "Foo \\u0060", "Foo `" },
```

Comment:
I didn't check all the source code, but I mainly looked at the tests.

I think these would work but is it worth trying some combinations involved unicode for backslash?
eg ```"\u005Du0060", "`"```

Reply:
Excellent catch.